challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map{|word| word.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select{|word| word=='dog'}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.select{|word| word!='dog'}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.keep_if{|word| word=~/[cd][a-z]+/}
```

Comment:
This works, but there's an Enumerable method that is explicitly for pattern-matching.